src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -887,6 +850,81 @@ private void DeleteBlocks(Hash256 deletePointer)
             }
         }
 
+        public IEnumerable<Block> DeleteBlocksBeforeTimestamp(ulong cutoffTimestamp, CancellationToken cancellationToken)
```

Comment:
Please move this method outside also. Timestamp based cutoff may not apply to all case, eg: portal. Also, plaese add a check so that no block is deleted after or at `BlockTree.SyncPivot`.